mmv1/products/discoveryengine/ChatEngine.yaml

```
@@ -0,0 +1,175 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'ChatEngine'
```

It looks like this API resource is actually just `Engine` and can have different subfields to determine the type: `similarDocumentsConfig`, `chatEngineConfig`, or `mediaRecommendationEngineConfig`. Generally we represent API resources like this as a single Terraform resource. Were you specifically wanting to propose having them be separate?

```suggestion
name: 'Engine'
```

---

@melinath We also another PR for SearchEngine. We didn't want to create confusion between 2 PRs so we named it as ChatEngine and SearchEngine

---

Yes that's intended, this resource is implemented to only manage Engines with solution_type = `SOLUTION_TYPE_CHAT` , the reason for such design is that depending on the `solution_type`, the functionality/resource behavior significantly varies as well as schema. 
`SOLUTION_TYPE_CHAT` engines does not use `similarDocumentsConfig` or `mediaRecommendationEngineConfig`, if we were to use a single resource in MMv1, we have to set all three `*Config` fields to optional. This implementation ensures each required fields can be validated.

---

`SOLUTION _TYPE_SEARCH` Engine is in [this PR](https://github.com/GoogleCloudPlatform/magic-modules/pull/9822)